src/main/kotlin/lotto/LottoMachine.kt

```diff
@@ -0,0 +1,20 @@
+package lotto
+
+class LottoMachine(private val receipt: Receipt) {
+    fun issue(): List<Lotto> {
+        return List(receipt.lottoCount) { Lotto { generateNumbers(Lotto.LOTTO_NUMBER_COUNT) } }
+    }
+
+    fun verify(lastNumber: String, lottos: List<Lotto>): List<StatResult> {
+        val checker = Checker(lastNumber)
+        return Stat(lottos, checker).sumRecords
+    }
+
+    fun yields(resultList: List<StatResult>): Double {
+        val totalProfit = resultList.sumOf { it.sum }
+        return Profit(
+            profit = totalProfit.toDouble(),
+            payment = receipt.payment.toDouble()
+        ).yields
```

Comment:
다음 단계에 제시된 일급컬렉션을 활용한다면 이 책임을 옮겨볼 수도 있을것 같아요. 꼭 이대로 하실 필요는 없으시고 다만 `LottoMachine` 이 `Profit` 을 만드는게 어색해 보인다는 느낌은 있습니다.
```suggestion
// List<StatResult> 의 일급 컬랙션타입이 StatResults 라고 한다면...
        return statResults.profit(receipt).yields
```
그런데 엄밀하게 말하면 수익률은 지출과 수입만으로 계산하면 되므로 로또를 사고난 잔액이 포함되어 있는 최초 금액(`receipt#payment`)는 필요 없고 `List<MatchState>` 또는 `List<StatResult>` 만으로 계산해야 합니다.